homeassistant/components/sensor/wunderground.py

```diff
@@ -0,0 +1,141 @@
+"""Support for Wunderground weather service.
+from datetime import timedelta
+import logging
+import requests
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.const import CONF_API_KEY
+from homeassistant.const import TEMP_FAHRENHEIT
+from homeassistant.const import TEMP_CELSIUS
+
+CONF_PWS_ID = 'pws_id'
+_URLCONST = '/conditions/q/pws:'
+_RESOURCE = 'http://api.wunderground.com/api/'
+_LOGGER = logging.getLogger(__name__)
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+
+# Sensor types are defined like: Name, units
+SENSOR_TYPES = {
+    'weather': ['Weather Summary', None],
+    'station_id': ['Station ID', None],
+    'feelslike_c': ['Feels Like (°C)', TEMP_CELSIUS],
+    'feelslike_f': ['Feels Like (°F)', TEMP_FAHRENHEIT],
+    'feelslike_string': ['Feels Like', None],
+    'heat_index_c': ['Dewpoint (°C)', TEMP_CELSIUS],
+    'heat_index_f': ['Dewpoint (°F)', TEMP_FAHRENHEIT],
+    'heat_index_string': ['Heat Index Summary', None],
+    'dewpoint_c': ['Dewpoint (°C)', TEMP_CELSIUS],
+    'dewpoint_f': ['Dewpoint (°F)', TEMP_FAHRENHEIT],
+    'dewpoint_string': ['Dewpoint Summary', None],
+    'wind_kph': ['Wind Speed', 'kpH'],
+    'wind_mph': ['Wind Speed', 'mpH'],
+    'UV': ['UV', None],
+    'pressure_in': ['Pressure', 'in'],
+    'pressure_mb': ['Pressure', 'mbar'],
+    'wind_dir': ['Wind Direction', None],
+    'wind_string': ['Wind Summary', None],
+    'temp_c': ['Temperature (°C)', TEMP_CELSIUS],
+    'temp_f': ['Temperature (°F)', TEMP_FAHRENHEIT],
+    'relative_humidity': ['Relative Humidity', '%'],
+    'visibility_mi': ['Visibility (miles)', 'mi'],
+    'visibility_km': ['Visibility (km)', 'km'],
+    'precip_today_in': ['Precipation Today', 'in'],
+    'precip_today_metric': ['Precipitation Today', 'mm'],
+    'precip_today_string': ['Precipitation today', None],
+    'solarradiation': ['Solar Radiation', None]
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Wunderground sensor.
+    payload = config.get('payload', None)
+    rest = WUndergroundData(_RESOURCE,
+                            config.get(CONF_PWS_ID),
+                            config.get(CONF_API_KEY),
+                            payload)
```

Comment:
Please use `voluptuous` to ensure that the API key and the pws ID are provided by the user.